fw/hpack.c

```diff
@@ -1324,6 +1324,7 @@ tfw_hpack_hdr_set(TfwHPack *__restrict hp, TfwHttpReq *__restrict req,
 		break;
 	case TFW_TAG_HDR_COOKIE:
 		parser->_hdr_tag = TFW_HTTP_HDR_COOKIE;
+		h2_set_hdr_cookie(req, &entry->cstate);
```

Comment:
This is subtle place. If I right here, might be inaccuracy in some cases. Let's assume that in the first request we received 1 cookie header, then in the second request we received 3 new headers and plus one indexed header from the last request. We calculate `cookie_num` for 3 headers and after override it with old one `cookie_num` that was stored in hpack entry. I'm not sure that we must fix it, but if it works so we definitely must write a comment.

Reply:
Good point, I will add test and fix